roachtest: election-after-restart failed

SHA: https://github.com/cockroachdb/cockroach/commits/e6348bb4abbfd117424c382ce5ab42e8abbe88f0

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stressrace TESTS=election-after-restart PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=974325&tab=buildLog

```
The test failed on release-2.1:
	test.go:611,election.go:67: Error: cannot dial server.
		Is the server running?
		If the server is running, check --host client-side and --advertise server-side.
		
		read tcp 127.0.0.1:43156->127.0.0.1:26257: i/o timeout
		Failed running "sql"
		Error:  exit status 1
		
		
		/home/agent/work/.go/bin/roachprod run teamcity-974325-election-after-restart:1 -- ./cockroach sql --insecure -e "
		SET TRACING = on;
		SELECT * FROM test.kv;
		SET TRACING = off;
		SHOW TRACE FOR SESSION;
		": exit status 1

```

--- a/pkg/cmd/roachtest/election.go
+++ b/pkg/cmd/roachtest/election.go
@@ -57,7 +57,11 @@ func registerElectionAfterRestart(r *registry) {
 			// this up are working (we trigger elections eagerly, but not so
 			// eagerly that multiple elections conflict with each other).
 			start = timeutil.Now()
-			buf, err := c.RunWithBuffer(ctx, t.l, c.Node(1), `./cockroach sql --insecure -e "
+			// Use a large CONNECT_TIMEOUT so that if the initial connection
+			// takes ages (perhaps due to some cli-internal query taking a
+			// very long time), we fail with the duration check below and
+			// not an opaque error from the cli.
+			buf, err := c.RunWithBuffer(ctx, t.l, c.Node(1), `COCKROACH_CONNECT_TIMEOUT=240 ./cockroach sql --insecure -e "
 SET TRACING = on;
 SELECT * FROM test.kv;
 SET TRACING = off;

